### PR TITLE
Added optional dependencies for the backends.

### PR DESCRIPTION
**Pull request by gbin**

### CODE REVIEW

**Comment by gbin** [comment]
Fixes #575

File: setup.py [hunk 1]
```diff
@@ -120,6 +120,14 @@ def read(fname):
         package_data={
             '': ['*.txt', '*.rst', '*.plug', '*.md'],
         },
+        extras_require={
+            'graphic':  ['PySide', ],
+            'hipchat': ['hypchat', 'sleekxmpp', ],
+            'IRC': ['irc', ],
+            'slack': ['slackclient', ],
+            'telegram': ['python-telegram-bot', ],
+            'XMPP': ['sleekxmpp', ],
```
Comment (zoni) [inline setup.py]: sleekxmpp needs the `pyasn1` and `pyasn1-modules` in order to do certificate validations but doesn't have a hard dependency on them, so we should add those too.

File: setup.py [hunk 2]
```diff
@@ -120,6 +120,14 @@ def read(fname):
         package_data={
             '': ['*.txt', '*.rst', '*.plug', '*.md'],
         },
+        extras_require={
+            'graphic':  ['PySide', ],
+            'hipchat': ['hypchat', 'sleekxmpp', ],
```
Comment (zoni) [inline setup.py]: sleekxmpp needs the `pyasn1` and `pyasn1-modules` in order to do certificate validations but doesn't have a hard dependency on them, so we should add those too.

File: setup.py (hunk 1 above)
Comment (gbin) [inline setup.py]: Done

**Comment by zoni** [comment]
Gets my :+1:

**Pull request merged**